tasks_week03_Ruben_Chiche/include/Animal.h

```diff
@@ -0,0 +1,39 @@
+#pragma once
+#include <string>
+#include <vector>
+#include <iostream>
+#include "Location.h"
+
+using std::string;
+
+constexpr auto ROWS = 20;
+constexpr auto COLUMNS = 40;
+
+
+class Animal
+{
+public:
+	// constructor
+	Animal(const string&, const Location&);
+
+	// pure virtual methods
+	virtual char getInitial() const = 0;
+	virtual void step() = 0;
+	virtual void turnVertically() = 0;
+	virtual void turnHorizontally() = 0;
+	virtual void move() = 0;
+
+	// public methods
+	void printDetails() const;
```

Comment:
these should be virtual tooo

Reply:
Why ? i have access to the type of the animal with this code "typeid(*this).name()", so i can make this function general for all the animals.

Reply:
if it's printing the derived animal details. then you are right